pallets/schema/src/lib.rs

```diff
@@ -151,77 +151,66 @@ pub mod pallet {
-		#[pallet::weight(30_000 + T::DbWeight::get().reads_writes(3, 2))]
-		pub fn authorise(
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::delegate())]
+		pub fn delegate(
 			origin: OriginFor<T>,
-			auth: SchemaParams<T>,
+			tx_schema: SchemaParams<T>,
 			delegates: Vec<CordAccountOf<T>>,
 			tx_signature: SignatureOf<T>,
 		) -> DispatchResult {
 			<T as Config>::EnsureOrigin::ensure_origin(origin)?;
 
 			ensure!(
-				!<SchemaHashes<T>>::contains_key(&auth.schema.digest),
+				!<SchemaHashes<T>>::contains_key(&tx_schema.schema.digest),
 				Error::<T>::InvalidTransactionHash
 			);
 
 			ensure!(
-				tx_signature.verify(&(&auth.schema.digest).encode()[..], &auth.schema.author),
+				tx_signature
+					.verify(&(&tx_schema.schema.digest).encode()[..], &tx_schema.schema.controller),
 				Error::<T>::InvalidSignature
 			);
 
-			ss58identifier::from_known_format(&auth.identifier, SCHEMA_IDENTIFIER_PREFIX)
+			ss58identifier::from_known_format(&tx_schema.identifier, SCHEMA_PREFIX)
 				.map_err(|_| Error::<T>::InvalidSchemaIdentifier)?;
 
 			let schema_details =
-				<Schemas<T>>::get(&auth.identifier).ok_or(Error::<T>::SchemaNotFound)?;
-			ensure!(!schema_details.revoked, Error::<T>::SchemaRevoked);
-
+				<Schemas<T>>::get(&tx_schema.identifier).ok_or(Error::<T>::SchemaNotFound)?;
 			ensure!(!schema_details.revoked, Error::<T>::SchemaRevoked);
 
-			if let Some(space) = auth.schema.space {
-				ensure!(
-					schema_details.schema.space == Some(space.clone()),
-					Error::<T>::SchemaSpaceMismatch
-				);
-
-				if schema_details.schema.author != auth.schema.author {
-					pallet_space::SpaceDetails::<T>::from_space_identities(
-						&space,
-						auth.schema.author.clone(),
-					)
-					.map_err(<pallet_space::Error<T>>::from)?;
-				}
-			} else {
-				ensure!(
-					schema_details.schema.author == auth.schema.author,
-					Error::<T>::UnauthorizedDelegation
-				);
-			}
-
-			Delegations::<T>::try_mutate(auth.identifier.clone(), |ref mut delegation| {
-				ensure!(
-					delegation.len() + delegates.len() <= T::MaxSchemaDelegates::get() as usize,
-					Error::<T>::TooManyDelegates
-				);
-				for delegate in delegates {
-					delegation
-						.try_push(delegate)
-						.expect("delegates length is less than T::MaxSchemaDelegates; qed");
-				}
-
-				<SchemaHashes<T>>::insert(&auth.schema.digest, &auth.identifier);
-
-				Self::deposit_event(Event::AddDelegates {
-					identifier: auth.identifier,
-					digest: auth.schema.digest,
-					author: auth.schema.author,
-				});
-				Ok(())
-			})
+			SchemaDetails::from_schema_delegates(
+				&tx_schema.identifier,
+				schema_details.schema.controller,
+				tx_schema.schema.controller.clone(),
+			)
+			.map_err(<Error<T>>::from)?;
+
+			SchemaDelegations::<T>::try_mutate(
+				tx_schema.identifier.clone(),
+				|ref mut delegation| {
+					ensure!(
+						delegation.len() + delegates.len() <= T::MaxSchemaDelegates::get() as usize,
+						Error::<T>::TooManyDelegates
```

Comment:
Any hardcoded number here for 'Too Many"? Can we change this number in the future?

Reply:
Applicable to every pallet where we use `TooManyDelegates` error.